addons/agent_mirrorpeer_controller.go

```diff
@@ -373,6 +373,9 @@ func (r *MirrorPeerReconciler) toggleMirroring(ctx context.Context, storageClust
 	}
 
 	// Determine if mirroring should be enabled or disabled
+	if sc.Spec.Mirroring == nil {
+		sc.Spec.Mirroring = &ocsv1.MirroringSpec{}
+	}
```

Comment:
Can we initialize this only in case when we need to enable mirroring? i.e. move this after `if enabled {...`
In other cases, nil is fine.